WordPress/src/main/java/org/wordpress/android/viewmodel/pages/PageItemUiStateHelper.kt

```diff
@@ -117,6 +132,38 @@ class PageItemUploadProgressHelper @Inject constructor(
         }
     }
 
+    fun setupPageActions(listType: PageListType, pageId: LocalId, site: SiteModel): Set<Action> {
+        return when (listType) {
+            SCHEDULED, PUBLISHED -> mutableSetOf(VIEW_PAGE, SET_PARENT, MOVE_TO_DRAFT, MOVE_TO_TRASH).apply {
+                if (canCancelPendingAutoUpload(pageId, site)) {
+                    add(CANCEL_AUTO_UPLOAD)
+                }
+            }
+            DRAFTS -> mutableSetOf(VIEW_PAGE, SET_PARENT, PUBLISH_NOW, MOVE_TO_TRASH).apply {
+                if (canCancelPendingAutoUpload(pageId, site)) {
+                    add(CANCEL_AUTO_UPLOAD)
+                }
+            }
+            TRASHED -> setOf(MOVE_TO_DRAFT, DELETE_PERMANENTLY)
+        }
+    }
+
+    @VisibleForTesting(otherwise = PRIVATE)
+    fun canCancelPendingAutoUpload(pageId: LocalId, site: SiteModel): Boolean {
+        val post = postStore.getPostByLocalPostId(pageId.value)
+
+        post?.let {
+            val uploadUiState = createUploadUiState(
```

Comment:
We might want to consider adding a TODO here as well. I think we'll want to pass the uploadState from the VM when https://github.com/wordpress-mobile/WordPress-Android/pull/11278 is merged. Wdyt?

Reply:
Yes, that's a really good idea. Thanks!

Reply:
Resolved here e036fa0